### PR TITLE
[1056] Ensure the `institution_country` is `GB` when a UK degree is selected

### PR DESCRIPTION
**Pull request by avinhurry**
## Context

When a UK degree selected the `institution_country` is `nil`, this is causing confusion, it should be `GB`.

A data migration will be merged to set these values for existing records in #8978, after this PR is merged.

## Changes proposed in this pull request

- Pass in GB through a hidden field in the view, similar to how the "Another flow" works.
- Remove the hardcoded "United Kingdom" string and replace with the default dynamic value.

## Guidance to review

The degree wizard could do with a refactor as part of a wider part of work, this is a quick fix to satisfy the requirements. 

## Link to Trello card

https://trello.com/c/jdPzTPQX/1056-bug-degree-institution-country-gb-showing-as-null

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [x] API release notes have been updated if necessary
- [x] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

**Comment by avinhurry** [comment]
[Bug: Degree institution country - GB showing as NULL](https://trello.com/c/jdPzTPQX/1056-bug-degree-institution-country-gb-showing-as-null)

File: spec/forms/candidate_interface/degree_wizard_spec.rb [hunk 1]
```diff
@@ -765,24 +765,6 @@
...
 
-    describe '#sanitize_country' do
-      let(:stored_data) { {}.to_json }
-      let(:attrs) { { uk_or_non_uk: 'uk', country: 'FR', current_step: :country } }
-
-      it 'clears country' do
-        wizard = described_class.new(store, attrs)
-        expect(wizard.country).to be_nil
-        expect(wizard.uk_or_non_uk).to eq 'uk'
-      end
```
Comment (JR-G) [inline spec/forms/candidate_interface/degree_wizard_spec.rb:776]: Just want to try and find the historic context as to why we were sanitising this in the first place!

**Comment by JR-G** [comment]
I've just tested switching from an international degree to a UK awarded one and it returns you to the review page with some of the values scrubbed, rather than taking you back through the flow (current implementation). Would you mind verifying this? Thanks! 🙏 
image

**Comment by avinhurry** [comment]
> I've just tested switching from an international degree to a UK awarded one and it returns you to the review page with some of the values scrubbed, rather than taking you back through the flow (current implementation). Would you mind verifying this? Thanks! 🙏 image

Great spot @JR-G!

It looks like this is an existing bug which previously only affected editing a degree when on support, there was a test which tested for the incorrect journey, [here](https://github.com/DFE-Digital/apply-for-teacher-training/pull/8956/commits/0b73db7e604c361ad6b03910645090d4cb951236). 

I have [simplified the `country_changed?`](https://github.com/DFE-Digital/apply-for-teacher-training/pull/8956/commits/919b64a9bf4ac2507560966c75c4570fea54d25b) method and [added a new feature test for it](https://github.com/DFE-Digital/apply-for-teacher-training/pull/8956/commits/a2bbf21cc14a0972cc1df34569d3f8a1d5e22946). It seems to have fixed the problem. 

Would you be able to give it another look please?

**Review by JR-G — APPROVED** [review]
Nice work & great investigating Avin! ⭐

**Pull request merged**